engine/crates/engine-v2/src/operation/modifier/query.rs

```diff
@@ -150,6 +153,51 @@ where
                         self.handle_modifier_resulted_in_error(modifier_id, modifier.impacted_fields, err);
                     }
                 }
+                // include and skip are handled separately
+                _ => {}
+            }
+
+            // as we make sure that every instance of a field creates at most one query modifier
+            // additional query modifiers guaranteed to be for different instances of the field.
+            // since we merge the field instances anyhow, any one instance of a field that should not be skipped is enough to include the field. (we use `all` below to avoid a case of no rules)
+            // within the same field instance, more than one value ID means the others are derived from parent fragments, so we should check if any of the values mark the field as skipped.
+            // (which is the inverse logic of multiple instances of the same field)
+            let skip_field = self
+                .operation
+                .query_modifiers
+                .iter()
+                .filter(|modifier| matches!(modifier.rule, QueryModifierRule::SkipInclude { .. }))
+                .all(|modifier| match &modifier.rule {
```

Comment:
would have used `filter_map(...).reduce(|(a, b)| a && b)`

Reply:
`any` will short circuit in this case, so it'd possibly be more efficient